Keep a fallback for the is_scoped_enum builtin missing in older g++.

--- a/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/__type_traits/is_enum.h
+++ b/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/__type_traits/is_enum.h
@@ -28,12 +28,30 @@
 
 #if _LIBCPP_STD_VER >= 23
 
+#if __has_builtin(__is_scoped_enum)
+
 template <class _Tp>
 struct _LIBCPP_NO_SPECIALIZATIONS is_scoped_enum : bool_constant<__is_scoped_enum(_Tp)> {};
 
 template <class _Tp>
 _LIBCPP_NO_SPECIALIZATIONS inline constexpr bool is_scoped_enum_v = __is_scoped_enum(_Tp);
 
+#else // __has_builtin(__is_scoped_enum)
+
+template <class _Tp, bool = is_enum_v<_Tp> >
+struct __is_scoped_enum_helper : false_type {};
+
+template <class _Tp>
+struct __is_scoped_enum_helper<_Tp, true> : public bool_constant<!is_convertible_v<_Tp, underlying_type_t<_Tp> > > {};
+
+template <class _Tp>
+struct _LIBCPP_NO_SPECIALIZATIONS is_scoped_enum : public __is_scoped_enum_helper<_Tp> {};
+
+template <class _Tp>
+_LIBCPP_NO_SPECIALIZATIONS inline constexpr bool is_scoped_enum_v = is_scoped_enum<_Tp>::value;
+
+#endif // __has_builtin(__is_scoped_enum)
+
 #endif // _LIBCPP_STD_VER >= 23
 
 _LIBCPP_END_NAMESPACE_STD
